File: algorithms/172.swift
// https://leetcode.com/problems/factorial-trailing-zeroes

class Solution {
    func trailingZeroes(n: Int) -> Int {
        var p = 1.0
        var zs = 0
        while Double(n)/pow(5.0, p) >= 1 {
            zs += Int(Double(n)/pow(5.0, p))
            p += 1.0
        }
        return zs
    }
}

File: algorithms/172.py
# https://leetcode.com/problems/factorial-trailing-zeroes


class Solution(object):
    def trailingZeroes(self, n):
        """
        :type n: int
        :rtype: int
        """
        p = 1
        zs = 0
        while n / (pow(5, p)) >= 1:
            zs += math.floor(n / pow(5, p))
            p += 1
        return int(zs)
